### DISCUSSION TITLE
Noob question (sorry) - how to set OpenRGB with CPU temps

### DISCUSSION THREAD

**Discussion by Gravity17** [jath03/openrgb-python#21] [Q&A]
Hi @jath03,

Just came across OpenRGB to replace NZXT Cam as it was causing BSODs.

I only used CAM NZXT to monitor my CPU Temps, and was hoping to replicate this functionality in OpenRGB. 

From the sounds of it this is possible, using the Python scripts in this repository, but I am a complete newcomer to Python, so don't really know where to begin.

I have installed OpenRGB and this has identified my RGBs (NZXT Smart Device V2 - these are the lights I want tied to the CPU temp - & X570 Aorus Ultra) and I am able to control the lighting of these by using the different GUI options within OpenRGB - but that is as far as I got (though I do see there is a setting for SDK Server - though not sure what I do other than turn it on).

I have installed Python 3.92 from Python.org - and it kinda goes downhill from there! ;-)

I have tried looking at the Python scripts, and the docs in the repository, but when I try to run them nothing happens (because I know I am doing something wrong, but I am not sure what).

Also do I need to tell the script which lights to turn on (e.g. NZXT Smart Device V2) and do I need to tell it which sensor - as if so, I would need help with that  too I'm afraid.

Is there a step-by-step guide on how to make this work - and apologies in advance if there is and I have missed it. If not no worries and sorry for taking up your time! :)

TIA!

**Comment by jath03** [comment]
Your main problem is that you're probably running into issue #20, which is my fault, not yours.  Once I get that fixed, and you learn the basics of the library from the [documentation](https://openrgb-python.readthedocs.io/en/latest/pages/usage.html), you can probably modify the [temperature example](https://github.com/jath03/openrgb-python/blob/master/examples/temperature/temperature.py) to work with your setup.  

Right now, that example is pretty personalized for my setup and not designed to be easily modifiable, but it shouldn't be too hard to make it do what you want.  I'll probably update the example to be more easily customizable soon, but if you don't want to wait, the main two things you'll have to change are the device you're setting ([lines 20-27](https://github.com/jath03/openrgb-python/blob/master/examples/temperature/temperature.py#L20-L27)) and the temperature sensor you're tracking ([lines 68-72](https://github.com/jath03/openrgb-python/blob/master/examples/temperature/temperature.py#L68-L79)).  The library I'm using for temperature only works on linux, so you could alternatively try OpenHardwareMonitor ([link](https://openhardwaremonitor.org/), [link](https://stackoverflow.com/questions/60692260/how-can-i-get-cpu-temperature-in-python-assuming-that-i-have-windows-32-bits)) for windows.  I'll see what I can do about a guide.

While I would pretty much always suggest going the DIY route, you could also try @inlart's [RGBmon](https://github.com/inlart/RGBmon) which would probably be easier to setup, though I don't think it supports temp monitoring on windows (yet).
